### PR TITLE
Stack visibility checkboxes

### PR DESCRIPTION
**Pull request by NightJar**
do not display the options to show or hide resource fields in a row,
rather display them in a column/stack as per the designs.

image

### CODE REVIEW

File: src/Model/ResourceField.php [hunk 1]
```diff
@@ -59,12 +59,17 @@ public function getCMSFields()
             $fields->dataFieldByName('ReadableName')
                 ->setAttribute('placeholder', $this->Name);
 
-            $summary = $fields->dataFieldByName('ShowInSummaryView');
-            $detail = $fields->dataFieldByName('ShowInDetailView');
-            $duplicates = $fields->dataFieldByName('RemoveDuplicates');
+            $summary = $fields->dataFieldByName('ShowInSummaryView')
+                ->addExtraClass('VisibilityOptions__Option');
```
Comment (ScopeyNZ) [inline src/Model/ResourceField.php]: I mentioned this in another PR but convention is that CSS classes are all lower case. If we're not doing this already I think we should be from now on. In this case it should be `visibility-options__option`.

Comment (NightJar) [reply src/Model/ResourceField.php]: Thanks, I'll update.

Comment (NightJar) [reply src/Model/ResourceField.php]: Done :)

File: src/Model/ResourceField.php [hunk 2]
```diff
@@ -59,12 +59,17 @@ public function getCMSFields()
...
+                ->addExtraClass('visibility-options__option');
+            $detail = $fields->dataFieldByName('ShowInDetailView')
+                ->addExtraClass('visibility-options__option');
+            $duplicates = $fields->dataFieldByName('RemoveDuplicates')
+                ->addExtraClass('visibility-options__option');
 
             $fields->removeByName(['ShowInSummaryView', 'ShowInDetailView', 'RemoveDuplicates',]);
-            $fields->insertBefore(FieldGroup::create('Visibility', [$summary, $detail, $duplicates]), 'Order');
+            $visibilityOptions = FieldGroup::create('Visibility', [$summary, $detail, $duplicates])
+                ->addExtraClass('VisibilityOptions');
```
Comment (ScopeyNZ) [inline src/Model/ResourceField.php]:
```suggestion
                ->addExtraClass('visibility-options');
```

File: client/src/styles/VisibilityOptions.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.VisibilityOptions.field .visibility-options__option {
```
Comment (ScopeyNZ) [inline client/src/styles/VisibilityOptions.scss]:
```suggestion
.visibility-options.field .visibility-options__option {
```
My suggestion assumes that you have added the `.field` for specificity - if not can we get away with just `.visibility-options__option`?

Comment (NightJar) [reply client/src/styles/VisibilityOptions.scss]: Yeah it's for specificity unfortunately.
I'll double check if it's needed. I presume the dist file for this module will always load after the class it's overriding.

Comment (NightJar) [reply client/src/styles/VisibilityOptions.scss]: ... I've forgotten to lower case a class anyway, I'll check and fix both.

**Comment by NightJar** [comment]
Approved so merging :)

**Pull request merged**